lib/common/variadic/startingBlock.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package variadic
+
+import (
+	"encoding/binary"
+	"errors"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// StartingBlock represents an optional interface type (int,hash).
+type StartingBlock struct {
+	value interface{}
+}
+
+// NewStartingBlock returns a new optional.Uint32
+func NewStartingBlock(startingBlockMsg []byte) (*StartingBlock, error) {
```

Comment:
we should probably avoid using `Msg` here because this is within the generic `lib/common` package and this is taking `[]byte` not a network message type

Reply:
agree, could just call it `in` or `data` or something, not necessarily a msg

Reply:
done